Libs/Widgets/ctkPathLineEdit.cpp

```diff
@@ -38,6 +38,177 @@
 #include "ctkPathLineEdit.h"
 #include "ctkUtils.h"
 
+// QFileSystemModel is very resource-intensive (if too many are created then it can crash the application
+// on Linux and Mac), therefore we don't want to create one for each widget but share two models between all
+// widgets.
+//
+// If name filtering is set then the global file system model cannot be used and so a custom file system model
+// is used in these widgets.
+//
+// From Qt-5.14, there as an option to disable file system watching (QFileSystemModel::DontWatchForChanges).
+// If CTK's minimum required Qt version reaches Qt-5.14 then probably these global file system models will
+// not be needed.
+
```

Comment:
I suggest to add these into an  anonymous namespace.

```
namespace
{

[...]

} // end of anonymous namespace

Reply:
thank you, I'll add this